microceph/database/schema.go

```diff
@@ -46,3 +47,23 @@ CREATE TABLE services (
 
 	return err
 }
+
+// Adds client config table in database schema.
+func schemaUpdate2(ctx context.Context, tx *sql.Tx) error {
+	stmt := `
+CREATE TABLE client_config (
+  id                            INTEGER  PRIMARY KEY AUTOINCREMENT NOT NULL,
+  member_id                     INTEGER,
+  key                           TEXT     NOT  NULL,
+  value                         TEXT     NOT  NULL,
+  FOREIGN KEY (member_id) REFERENCES "internal_cluster_members" (id) ON DELETE CASCADE,
+  UNIQUE(member_id, key)
+);
+
+CREATE UNIQUE INDEX cc_index ON client_config(coalesce(member_id, 0), key);
```

Comment:
I believe declaring a col UNIQUE will cause sqlite to auto-create an index, is there a reason to create one explicitly?

Reply:
The Index with coalesce is used to remedy the NIL values present in the client_config table since SQLite does not treat a NIL value as unique. However, It is true that we do not need 2 indexes, therefore I have removed the UNIQUE(member_id, key) reference from the schema definition.